pylib/Stages/Reporter/TextFile.py

```diff
@@ -88,10 +88,16 @@ def execute(self, log, keyvals, testDef):
                     pass
                 if 0 != lg['status']:
                     try:
-                        print("\tERROR:",lg['stderr'], file=self.fh)
+                        print("\tERROR:", file=self.fh)
+                        if lg['stderr']:
```

Comment:
I don't think you need these `if` statements because the result will *always* be a list, right?  (although the list might be empty)

I think you can just go directly to the `for l in ...` statement.

Reply:
Actually, I think you need to restructure this a little.  You don't need the try blocks any more, and you want to check to see if you want to print the ERROR heading first.  Perhaps something like this:

```python
def _print_error_block(name, lines):
    if len(lines) > 0:
        print("\tERROR ({name})".format(name=name), file=self.fh)
        for l in lines:
            print(l, file=self.fh)

if 0 != lg['status']:
    _print_error_block(lg['stderr'])
    _print_error_block(lg['stdout'])
```

Reply:
I would also just add a `if 'stderr' in lg:` and similar for stdout. Otherwise, agreed with these points.
Also, mind the missing `name` arg 👍

Reply:
Hah ya; that's what one gets for typing code out in a web browser and not testing it.  😉